ui/packages/components/src/RerunButtonV2.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+import { Button } from '@inngest/components/Button';
+import { RiLoopLeftLine } from '@remixicon/react';
+
+import { cn } from './utils/classNames';
+
+type Props = {
+  onClick: () => Promise<unknown>;
+};
+
+export function RerunButton(props: Props) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function onClick() {
+    setIsLoading(true);
+    try {
+      await props.onClick();
+    } catch {
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  return (
+    <Button
+      btnAction={onClick}
+      disabled={isLoading}
+      icon={<RiLoopLeftLine className={cn(' text-sky-500', isLoading && 'animate-spin')} />}
```

Comment:
```suggestion
      isLoading={isLoading} />}
```

And the "Cancel" button should also not have the icon, like the figma file, it doesn't bring a lot of value and the icon has very low contrast